modules/fs.js

```diff
@@ -429,16 +471,21 @@ function listTree(path) {
  * └── test.txt
  */
 function removeTree(path) {
-    var file = resolveFile(path);
-    // do not follow symlinks
-    if (file.isDirectory() && !isLink(file.getPath())) {
-        for each (var child in file.list()) {
-            removeTree(join(file, child));
+    var nioPath = resolvePath(path);
+    Files.walkFileTree(nioPath, new JavaAdapter(java.nio.file.SimpleFileVisitor, {
+        visitFile: function (file, attrs) {
+            Files.delete(file);
+            return java.nio.file.FileVisitResult.CONTINUE;
+        },
+        postVisitDirectory: function (dir, e) {
+            if (e == null) {
+                Files.delete(dir);
+                return java.nio.file.FileVisitResult.CONTINUE;
+            } else {
+                throw e;
+            }
```

Comment:
Syntactic sugar: you can instantiate SimpleFileVisitor directly in ringo/rhino, passing a JS object containing the overriding methods, so you could do `new SimpleFileVisitor({"visitFile": ..., "postVisitDirectory": ...})` instead of `new JavaAdapter(...)`.